src/main/java/org/truffleruby/stdlib/psych/PsychParserNodes.java

```diff
@@ -166,28 +165,38 @@ private StreamReader newStreamReader(DynamicObject io) {
     @CoreMethod(names = "get_event", needsSelf = false, required = 1)
     public abstract static class GetEventNode extends CoreMethodArrayArgumentsNode {
         @TruffleBoundary
-        @Specialization
-        protected Event getEvent(Parser parser) {
-            return parser.getEvent();
+        @Specialization(guards = "isParser(parser)")
+        protected DynamicObject getEvent(DynamicObject parser) {
```

Comment:
I guess the usage of this node guarantees that the guard always matches?

Reply:
Yes, but I think it's still good to have the guard to prevent accidental misuses.

Reply:
Sure, if it is clear error, than it's fine.